File: JavaScript/Easy/arrays_hashing/imageSmoother.js
var solution = function(img){
    m = img.length
    n = img[0].length

    var gridHelp = (row, col) => {
        let total = 0
        let count = 0

        let top = Math.max(0, row-1)
        let bottom = Math.min(m, row + 2)
        let left = Math.max(0, col - 1)
        let right = Math.min(n, row + 2)

        for (let r = 0; r< row; r++){
            for (let c = 0; c<col; c++){
                total += img[r][c]
                count ++
            }
        }
    
        return Math.floor(total/count)

    }

    let output = new Array(m).fill(0).map(() => new Array(n).fill(0))

    for (let col = 0; col < n; col ++){
        for (let row = 0; row < m; row ++){
            output[row][col] = gridHelp(row, col)
        }
    }

    return output

}

File: Python/Easy/Arrays_Hashing/imageSmoother.py
import math
class Solution:
    def imageSmoother(self, img):
        m = len(img)
        n = len(img[0])

        def gridHelper(row, col):
            # set the total and count
            total, count = 0,0

            top = max(0, row-1)
            bottom = min(m, row+2)
            left = max(0, col - 1)
            right = min(n, col+2)

            for r in range(top, bottom):
                for c in range(left, right):
                    total += img[r][c]
                    count += 1

            return total // count
        
        return [[gridHelper(row, col) for row in m] for col in n]